### PR TITLE
Updating Bedrock Converse API citations format to handle web citations

### LINKED ISSUE #1323: [BUG] Bedrock Runtime API CitationLocation Tagged Union Structure Error
### Checks

- [x] I have updated to the lastest minor and patch version of Strands
- [x] I have checked the documentation and this is not expected behavior
- [x] I have searched [./issues](./issues?q=) and there are no duplicates of my issue

### Strands Version

1.19.0

### Python Version

3.14.2

### Operating System

N/A

### Installation Method

pip

### Minimum Reproducible code
```
from strands.models.bedrock import BedrockModel
import asyncio

async def processing_streaming_response():
    model = BedrockModel(
        model_id="us.anthropic.claude-3-7-sonnet-20250219-v1:0"
    )

    messages = [
        {
            "role": "user",
            "content": [
                {
                    "text": "Use this media in your response."
                }
            ]
        },
        # Assistant response with citations
        {
            "role": "assistant",
            "content": [
                {
                    "text": "I've analyzed the document. Here's what I found:"
                },
                {
                    "citationsContent": {
                        "citations": [
                            {
                                "location": {
                                    "documentChar": {
                                        "documentIndex": 0,
                                        "start": 150,
                                        "end": 300
                                    }
                                },
                                "sourceContent": [
                                    {
                                        "text": "Employee benefits include health insurance and retirement plans"
                                    }
                                ],
                                "title": "Benefits Section"
                            },
                            {
                                "location": {
                                    "documentPage": {
                                        "documentIndex": 0,
                                        "start": 2,
                                        "end": 3
                                    }
                                },
                                "sourceContent": [
                                    {
                                        "text": "Vacation policy allows 15 days per year"
                                    }
                                ],
                                "title": "Vacation Policy"
                            }
                        ],
                        "content": [
                            {
                                "text": "Based on the document, employees receive comprehensive benefits including health insurance, retirement plans, and 15 days of vacation per year."
                            }
                        ]
                    }
                }
            ]
        }
    ]

    #model._format_bedrock_messages(messages=messages)
    response_text = ""
    async for chunk in model.stream(
        messages=messages
    ):
        #response_text += chunk
        print(chunk, end="", flush=True)

    return response_text

asyncio.run(processing_streaming_response())
```

- Install strands through this command: `pip install strands-agents strands-tools`
- Run the code, and you will observe the following error:
```
botocore.exceptions.ParamValidationError: Parameter validation failed:
Must set one of the following keys for tagged unionstructure messages[1].content[1].citationsContent.citations[0].location: web, documentChar, documentPage, documentChunk, searchResultLocation.
Must set one of the following keys for tagged unionstructure messages[1].content[1].citationsContent.citations[1].location: web, documentChar, documentPage, documentChunk, searchResultLocation.
```

### Expected Behavior

- CitationLocation objects should preserve the tagged union structure required by AWS Bedrock Runtime API: https://docs.aws.amazon.com/bedrock/latest/APIReference/API_runtime_CitationLocation.html

### Actual Behavior

- It is not doing that but expecting all the keys should be present in the API. 

### Additional Context

_No response_

### Possible Solution

- Preserve the union structure when processing CitationLocation objects.
- Sample code below(need to test it out):
```
def _format_citation_location(location: dict) -> dict:
    """Preserve tagged union structure for AWS Bedrock API compatibility"""
    
    # Check which union type is present and preserve wrapper
    if "documentChar" in location:
        inner = location["documentChar"]
        filtered_inner = {}
        if "documentIndex" in inner:
            filtered_inner["documentIndex"] = inner["documentIndex"]
        if "start" in inner:
            filtered_inner["start"] = inner["start"]
        if "end" in inner:
            filtered_inner["end"] = inner["end"]
        return {"documentChar": filtered_inner}
    
    elif "documentPage" in location:
        # Similar processing for documentPage union type
        return {"documentPage": filtered_inner}
    
    # Handle other union types: documentChunk, searchResultLocation, web
    # ...
    
    else:
        # Fallback: infer union type from available fields
        if "url" in location:
            return {"web": {"url": location["url"]}}
        # Default to documentChar for document locations
        return {"documentChar": location}
```

### Related Issues

_No response_

### PR DESCRIPTION
**Pull request by danilop**
## Description

This PR fixes citation location handling in the Bedrock Converse API to properly preserve the tagged union structure required by the API.

**Problem:**
The Bedrock API requires `CitationLocation` to be a tagged union with exactly one wrapper key (`web`, `documentChar`, `documentPage`, `documentChunk`, or `searchResultLocation`). The previous implementation was flattening document-based citations, causing validation errors like:
```
botocore.exceptions.ParamValidationError: Must set one of the following keys for tagged union structure...
```

**Solution:**
- Refactored citation type definitions to properly model the tagged union structure with wrapper types (`DocumentCharLocation`, etc.) and inner content types (`DocumentCharLocationInner`, etc.)
- Added `_format_citation_location` helper method that preserves the union structure when filtering citation locations
- Added `SearchResultLocation` type for complete API coverage
- Added comprehensive tests for all 5 location types

**Changes:**
- `src/strands/types/citations.py`: Refactored types to model tagged union structure correctly
- `src/strands/models/bedrock.py`: Added `_format_citation_location` helper method
- `tests/strands/models/test_bedrock.py`: Added tests for all citation location types

## Related Issues

- Fixes #1323 (CitationLocation Tagged Union Structure Error)
- Blocking issue for #1154

## Documentation PR

N/A

## Type of Change

Bug fix

## Testing

- [X] I ran `hatch run prepare` (1616 tests pass)
- [X] All citation location types tested (web, documentChar, documentPage, documentChunk, searchResultLocation)
- [X] Edge cases tested (optional fields, invalid locations)

## Checklist
- [X] I have read the CONTRIBUTING document
- [X] I have added any necessary tests that prove my fix is effective or my feature works
- [X] I have added an appropriate example to the documentation to outline the feature, or no new docs are needed
- [X] My changes generate no new warnings
- [X] Any dependent changes have been merged and published

----

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/strands/types/citations.py [hunk 1]
```diff
@@ -18,67 +18,41 @@ class CitationsConfig(TypedDict):
...
     """
 
-    documentIndex: int
-    start: int
-    end: int
+    url: str
+    domain: str
 
 
-class DocumentChunkLocation(TypedDict, total=False):
```
Comment (dbschmigelski) [inline src/strands/types/citations.py:41]: Hi, we have to preserve backwards compatibility. So we cannot simply remove these classes.

These are typeddicts, so I see the argument that it would not be possible from someone to leverage
`CitationLocation = Union[DocumentCharLocation, DocumentChunkLocation, DocumentPageLocation]` to do instanceof checks in the first place. But this still needs to be workshopped to avoid breaking changes

**Review by dbschmigelski — CHANGES_REQUESTED** [review]

**Comment by kpx-dev** [comment]
hi team, any update on this? One of my customer is also looking forward for this feature support to test out Nova Web Grounding.
